src/witness.rs

```diff
@@ -61,20 +61,15 @@ impl Witness {
                 .collect();
 
             // TODO not every receipt should update resolver.
-            for rct in receipts.iter() {
-                match self.witness.get_state_for_prefix(&rct.body.event.prefix)? {
-                    // update key state in resolver
-                    Some(state) => {
-                        let resolver = self.resolvers.first().expect("There's no resolver set");
-                        // TODO: Should send signed event data as octet-stream
-                        if let Err(e) =
-                            ureq::post(&[resolver, "/messages/", &state.prefix.to_str()].join(""))
-                                .send_json(&state)
-                        {
-                            println!("Problem with publishing state in resolver: {:?}", e);
-                        };
-                    }
-                    None => (),
+            for receipt in receipts.iter() {
```

Comment:
It seems to me that sending kel for each receipt is redundant. Sending it once for each updated identifier (without repetitions) should have the same outcome.
By updated identifier I mean the receipt event's `i` field.